### PR TITLE
separate crates

### LINKED ISSUE #91: separate crates for lib and bin
creating this new issue to track things as discussed in #89 we can reduce dependencies for crates using lib by separating the current crate into two crates:
- blocks_iterator_cli (which contains the current main.rs and all other examples_
- blocks_iterator (the library which can be imported and the config struct will be put behind cli feature flag)

### PR DESCRIPTION
**Pull request by king-11**
fixed #91

### CODE REVIEW

**Comment by king-11** [comment]
ready for review @RCasatta @dpc

File: lib/src/lib.rs [hunk 1]
```diff
@@ -157,15 +157,14 @@ mod inner_test {
     use crate::bitcoin::Network;
     use crate::{iterate, Config};
     use std::sync::mpsc::sync_channel;
+    use test_log::test;
```
Comment (RCasatta) [inline lib/src/lib.rs:160]: didn't know about this, it's great!
Ideally would have been better to introduce this in a separate commit

File: lib/src/config.rs [hunk 2]
```diff
@@ -1,37 +1,39 @@
...
     /// however make it impossible calculate fees or access tx input previous scripts
-    #[arg(short, long)]
+    #[cfg_attr(feature = "clap", arg(short, long))]
     pub skip_prevout: bool,
 
     /// Maximum length of a reorg allowed, during reordering send block to the next step only
     /// if it has `max_reorg` following blocks. Higher is more conservative, while lower faster.
     /// When parsing testnet blocks, it may be necessary to increase this a lot
-    #[arg(short, long, default_value = "6")]
+    #[cfg_attr(feature = "clap", arg(short, long, default_value = "6"))]
```
Comment (RCasatta) [inline lib/src/config.rs:26]: This is great to do now that the crate is separate, but ideally this should have been a move only commit, followed by changes like this one.

**Review by RCasatta — APPROVED** [review]
ACK 5fdab285e56781b6759f214e282f404a3f35d130

There are some warning to be fixed on nix, can be fixed in next PR, this major refactor looks great, thanks!

**Pull request merged**